addons/web_editor/static/src/scss/web_editor.common.scss

```diff
@@ -1013,3 +1013,37 @@ section, .oe_img_bg, [data-oe-shape-data] {
         }
     }
 }
+
+// GRID
+.o_grid_mode {
+    @include media-breakpoint-up(md) {
+        display: grid !important;
+        grid-auto-rows: 50px;
+        grid-template-columns: repeat(12, 1fr);
+        row-gap: 0px;
+        column-gap: 0px;
+    }
+    --gutter-x: 0px;
+    --grid-item-padding-y: 10px;
+    --grid-item-padding-x: 10px;
+
+    > * {
+        width: 100%;
+        margin: 0;
+        padding: var(--grid-item-padding-y) var(--grid-item-padding-x);
+    }
+}
+
+.container-fluid > .o_grid_mode {
+    --gutter-x: 30px;
+}
+
+.o_grid_item {
+    word-break: break-word;
+
+    > img {
+        width: 100%;
+        height: 100%;
+        object-fit: cover !important;
+    }
+}
```

Comment:
I am worried about that rule, does this mean that if I have a column with texts, that I add an image inside the text... then switch to grid mode, I'll have the image cover the text? :grimacing: Or is there some magic I missed?

Reply:
The magic is that when adding an image in a text, it goes inside a `<p>` so it is not concerned by this rule.
However, there is a "problem" when we want to add text/image in an image (since the base image is the direct child of the div) but this problem "just" consists in the overlay not being able to cover all the content (since the image height is 100%) so there is always some overflow... Also the added image will not be `object-fit: cover` since it is not a direct child of the div.
I don't know how to fix that to be honest but I think that if someone wants to add an image, he should use the "Add element" option.

Reply:
> The magic is that when adding an image in a text, it goes inside a <p>

We cannot (or at least should not) rely on that: the editor possibilities can evolve but users can also copy/paste arbitrary HTML anywhere they want. If possible we should thus support the case where the image is a direct child and we do not want it to be covered. Easiest solution probably: when adding the o_grid_item class, check for direct child images who are alone in their column and add a `o_grid_item_image` class on them and target this in CSS, something like that :+1: 

I'll leave this conversation open, we can do that post merge.

Reply:
>  think that if someone wants to add an image, he should use the "Add element" option

Same remark: we cannot rely on that:
- If there are multiple ways to add an image via the UI and only one should be used... then let's hide all other ones.
- Even if there is only one button, user can still add arbitrary HTML (we cannot handle every way they can break stuff but it should be the most robust possible).